### DISCUSSION TITLE
Refer asset path inside of javascript file

### DISCUSSION THREAD

**Discussion by rafaeldev** [rails/jsbundling-rails#215] [Q&A]
I have a servicer worker where I have a list of files to be cached. Previous, I ran using `.erb` extension to get that, like this:
```js
const CACHE_VERSION = 'v2';
const CACHE_NAME = CACHE_VERSION + ':sw-cache-';

function onInstall(event) {
  console.log('[Serviceworker]', "Installing!", event);

  event.waitUntil(
    caches.open(CACHE_NAME).then(function prefill(cache) {
      return cache.addAll([
        '<%= asset_path "application.js" %>',

        '<%= asset_path "application.css" %>',

        '/offline.html',
      ]);
    })
  );
}
```

Now I decided to move `sw.js` to esbuild build flow but I don't know how can I asset `application.js` file link there.

Thanks!

**Comment by risen** [comment]
I also would like a way to get the asset_path in javascript.

For example, when I want to use [customize the icon library for shoelace.style](https://shoelace.style/components/icon#customizing-the-default-library), I can do this:

```javascript
registerIconLibrary("default", { resolver: (name) => `https://cdn.jsdelivr.net/npm/bootstrap-icons@1.11.0/icons/${name}.svg` })
```

… but I can't refer to local assets and do something like:

```javascript
registerIconLibrary("default", { resolver: (name) => asset_path(`icons/${name}.svg`) })
```

**Comment by seanharmer** [comment]
I've just managed to get something like this working. You can use the new RAILS_ASSET_URL helper from propshaft. To use it do something like this:
```
const particleTexturePath = RAILS_ASSET_URL("/particles/9.png")
console.log(particleTexturePath)
```

which outputs the full finger-printed path: `/assets/particles/9-61a18f3e.png`

If the path fails to resolve, there will be a log message in the rails server logs.

**Reply by risen** [reply]
Perfect, thanks for the heads up!
